### PR TITLE
Change how Select is displayed in circuit drawer

### PR DESCRIPTION
**Pull request by DSGuala**
```
0: ─╭●──────────┤  State
1: ─├●──────────┤  State
2: ─├Select(M0)─┤  State
3: ─╰Select(M0)─┤  State
M0 = 
[PauliX(wires=[2]) PauliX(wires=[3]) PauliY(wires=[2]) SWAP(wires=[2, 3])]
```

Will become:

```
0: ─╭Select(M0)─┤  State
1: ─├Select(M0)─┤  State
2: ─├Select(M0)─┤  State
3: ─╰Select(M0)─┤  State
M0 = 
[PauliX(wires=[2]) PauliX(wires=[3]) PauliY(wires=[2]) SWAP(wires=[2, 3])]
```

This is will change the arguments for `qml.Select()`: `qml.Select(ops, control_wires=[1,2])` will become `qml.Select(ops, control=[1,2])`

Additionally, the `control_wires` property of the operation will become `control`.
`qml.Select(...).control_wires` -> `qml.Select(...).control`

### CODE REVIEW

File: pennylane/templates/subroutines/select.py [hunk 1]
```diff
@@ -167,9 +166,9 @@ def ops(self):
         return self.hyperparameters["ops"]
 
     @property
-    def control_wires(self):
+    def control(self):
         """The control wires."""
-        return self.hyperparameters["control_wires"]
+        return self.hyperparameters["control"]
```
Comment (trbromley) [inline pennylane/templates/subroutines/select.py:171]: Do we need this property?

**Review by trbromley — COMMENTED** [review]
Thanks @DSGuala! Overall this makes sense to me.

**Comment by albi3ro** [comment]
How is it registering the list of operations as data?

**Comment by albi3ro** [comment]
@DSGuala Also, should we consider this a bugfix and merge it into the rc branch?

File: pennylane/templates/subroutines/select.py (hunk 1 above)
Comment (DSGuala) [reply pennylane/templates/subroutines/select.py:171]: It's not mandatory. We can remove it. Users could do `op.hyperparameters['control']`
Is that the preference?

**Review by Jaybsoni — APPROVED** [review]
I don't want to be a blocker so approving for now, but should we also update the example in the changelog if it uses `control_wires` instead of `control` [edited]

**Review by soranjh — COMMENTED** [review]
Why the base branch is `rc_2023-08-23-02-51-22`?

**Comment by DSGuala** [comment]
> How is it registering the list of operations as data?

The `Operator` parent class stores this into data from the input parameters. That is in [this line](https://github.com/PennyLaneAI/pennylane/blob/92d95035b0a1439cb19dc35e54ce4f91bc935aa8/pennylane/operation.py#L1073) of PennyLane. Is that what you were asking @albi3ro ?

**Comment by DSGuala** [comment]
> I don't want to be a blocker so approving for now, but should we also update the example in the doc-string if it uses `control_wires` instead of `control`

Thought this was already done :sweat: , will update. @Jaybsoni

File: pennylane/templates/subroutines/select.py (hunk 1 above)
Comment (trbromley) [reply pennylane/templates/subroutines/select.py:171]: I don't have a strong preference, but curious to see what @albi3ro thinks.

File: doc/development/release_notes.md
Comment (soranjh) [inline doc/development/release_notes.md:1]: Please be careful with debasing. There are commits picked up from other branches.

File: doc/releases/changelog-dev.md
Comment (soranjh) [inline doc/releases/changelog-dev.md:1]: Please be careful with debasing. There are commits picked up from other branches.

File: pennylane/_version.py
Comment (soranjh) [inline pennylane/_version.py:1]: Please be careful with debasing. There are commits picked up from other branches.

**Pull request merged**